internal-packages/workflow-designer-ui/src/settings/github-integration/github-integration-setting-form.tsx

```diff
@@ -42,9 +78,72 @@ function Installed({
 }: { repositories: GitHubIntegrationRepository[] }) {
 	const { data: workspace } = useWorkflowDesigner();
 	const { isLoading, data, handleSubmit } = useGitHubIntegrationSetting();
+	const [selectedTrigger, setSelectedTrigger] = useState<
+		WorkspaceGitHubIntegrationTrigger | undefined
+	>(data?.event);
+	const [callsign, setCallsign] = useState(data?.callsign || "");
+	const [selectedNextAction, setSelectedNextAction] = useState<
+		WorkspaceGitHubNextIntegrationAction | undefined
+	>(data?.nextAction);
+	const [payloadMaps, setPayloadMaps] = useState<
+		WorkspaceGitHubIntegrationPayloadNodeMap[]
+	>(data?.payloadMaps || []);
+
+	useEffect(() => {
+		if (data) {
+			setSelectedTrigger(data.event);
+			setCallsign(data.callsign || "");
+			setSelectedNextAction(data.nextAction);
+			setPayloadMaps(data.payloadMaps || []);
+		}
+	}, [data]);
+
 	if (isLoading) {
 		return null;
 	}
+
+	const getAvailablePayloadFields = (
+		trigger: WorkspaceGitHubIntegrationTrigger,
+	) => {
+		const fields = Object.values(WorkspaceGitHubIntegrationPayloadField.Enum);
+		const triggerParts = trigger.split(".");
+		const triggerPrefix = `${triggerParts[0]}.${triggerParts[1]}`;
+		return fields.filter((field) => field.startsWith(triggerPrefix));
+	};
+
+	const getAvailableNextActions = (
+		trigger: WorkspaceGitHubIntegrationTrigger,
+	) => {
+		return TRIGGER_TO_ACTIONS[trigger] ?? [];
+	};
+
+	const handleTriggerChange = (value: string) => {
+		const newTrigger = WorkspaceGitHubIntegrationTrigger.parse(value);
+		const currentTrigger = selectedTrigger;
+		setSelectedTrigger(newTrigger);
+
+		if (newTrigger !== currentTrigger) {
+			if (!requiresCallsign(newTrigger)) {
+				setCallsign("");
+			}
+
+			const availableActions = getAvailableNextActions(newTrigger);
+			if (
+				selectedNextAction &&
+				!availableActions.includes(selectedNextAction)
+			) {
+				setSelectedNextAction(undefined);
+			}
+		}
+	};
```

Comment:
Consider using useCallback for this definition. It could improve performance by preventing unnecessary re-renders